File: Lab2_qn2.py
#dot product and cross product

vector1=[5,6,1]
vector2=[2,8,9]

#Compute addition

i=0
sum=0
sum1=0
sum2=0
sum3=0
sumi=0
while i<=2:
    if i==0:
        sum=vector1[i]+vector2[i]
        sum1=sum
    elif i==1:
        sum=vector1[i]+vector2[i]
        sum2=sum
    else:
        sum = vector1[i] + vector2[i]
        sum3=sum
    i=i+1

summation=[sum1,sum2,sum3]
print(summation)


#Compute Dot product
j=0
sumdot=0
while j<=2:
    dot=vector1[j]*vector2[j]
    sumdot=sumdot + dot
    j=j+1

print(sumdot)

File: Lab2_Qn2.cpp
//Addition and dot product of two vectors

#include<iostream>
using namespace std;
int main()
{
    int A[]={1,2,3};
    int B[]={4,5,6};
    
    //define and compute addition
    int sum[3];
    int i=0;
    while (i<=2)
    {
        sum[i]=A[i]+B[i];
        //cout<<sum[i]<<endl;
        i=i+1;
    }
    //Printing the sum
    int j=0;
    cout<<"The resultant added vector is: "<<endl;
    
    while (j<=2)
    {
        cout<<sum[j]<<endl;
        j=j+1;
    }
    
    
    //Define and compute dot product
    int k=0;
    int dot=0;
    while (k<=2)
    {
        dot=dot+(A[k]*B[k]);
        k=k+1;
    }
    cout<<endl;
    cout<<"The dot product of the two vectors is: "<<dot<<endl;
        
    
    
    return 0;
}

/*The resultant added vector is: 
5
7
9

The dot product of the two vectors is: 32


*/
    


